File: sample/src/main/java/vn/tiki/sample/api/ApiService.java
package vn.tiki.sample.api;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import vn.tiki.sample.response.ListResponse;
import vn.tiki.sample.response.ProductResponse;

public interface ApiService {

  @SuppressWarnings("SameParameterValue") @GET("products")
  Observable<ListResponse<ProductResponse>> getProducts(
      @Query("page") int page,
      @Query("per_page") int perPage);
}

File: sample/src/main/java/vn/tiki/sample/api/ApiService.kt
package vn.tiki.sample.api

import io.reactivex.Single
import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiService {

  @GET("products")
  fun getProducts(
      @Query("page") page: Int,
      @Query("per_page") perPage: Int): Single<ResponseBody>
}
